driver-core/src/test/functional/com/mongodb/client/CommandMonitoringTestHelper.java

```diff
@@ -253,11 +254,27 @@ private static void massageActualCommand(final BsonDocument command, final BsonD
                 command.remove(key);
             } else if (command.isDocument(key) && expectedCommand.isDocument(key)) {
                 massageActualCommand(command.getDocument(key), expectedCommand.getDocument(key));
+            } else if (command.containsKey("pipeline") && expectedCommand.containsKey("pipeline")) {
+                massagePipeline(command, expectedCommand);
             }
         }
 
     }
 
+    // If the expected pipeline contains a $changeStream key with an empty document value, remove the
+    // startAtOperationTime and resumeAfter fields from the actual pipeline value.
+    private static void massagePipeline(final BsonDocument command, final BsonDocument expectedCommand) {
+        if (expectedCommand.containsKey("pipeline") && command.containsKey("pipeline")) {
+            BsonDocument expectedChangeStreamDocument = expectedCommand.getArray("pipeline").get(0).asDocument();
```

Comment:
Nit: It's possible that the pipeline could be an empty array (though perhaps no actual tests do that).  But if they did then `get(0)` will return null.  Safer to just add a check for a 0-length pipeline

Reply:
I'm surprised it doesn't fail against some of the existing tests, as there are a bunch with empty pipelines.  Perhaps this code path is not hit for those tests.

Reply:
I have added the check for a 0-length pipeline.